drivers/place/event_scrape.cr

```diff
@@ -0,0 +1,84 @@
+require "place_calendar"
+
+class Place::EventScrape < PlaceOS::Driver
+  descriptive_name "PlaceOS Event Scrape"
+  generic_name :EventScrape
+
+  default_settings({
+    zone_ids: ["placeos-zone-id"],
+    internal_domains: ["PlaceOS.com"]
+  })
+
+  accessor staff_api : StaffAPI_1
+
+  @zone_ids = [] of String
+  @internal_domains = [] of String
+
+  alias Event = PlaceCalendar::Event
+
+  struct SystemWithEvents
+    include JSON::Serializable
+
+    def initialize(@name : String, @zones : Array(String), @events : Array(Event))
+    end
+  end
+
+  def on_load
+    on_update
+  end
+
+  def on_update
+    @zone_ids = setting?(Array(String), :zone_ids) || [] of String
+    @internal_domains = setting?(Array(String), :internal_domains) || [] of String
+  end
+
+  def todays_bookings
+    response = {
+      internal_domains: @internal_domains,
+      systems: {} of String => SystemWithEvents
+    }
+
+    now = Time.local
+    start_epoch = now.at_beginning_of_day.to_unix
+    end_epoch = now.at_end_of_day.to_unix
+
+    @zone_ids.each do |z_id|
+      staff_api.systems(zone_id: z_id).get.as_a.each do |sys|
+        sys_id = sys["id"].as_s
+        # In case the same system is in multiple zones
+        next if response[:systems][sys_id]?
+
+        response[:systems][sys_id] = SystemWithEvents.new(
+          name: sys["name"].as_s,
+          zones: Array(String).from_json(sys["zones"].to_json),
+          events: get_system_bookings(sys_id, start_epoch, end_epoch)
+        )
+      end
+    end
+
+    response
+  end
+
+  def get_system_bookings(sys_id : String, start_epoch : Int64?, end_epoch : Int64?) : Array(Event)
+    booking_module = staff_api.modules_from_system(sys_id).get.as_a.find { |mod| mod["name"] == "Bookings" }
+    # If the system has a booking module with bookings
+    if booking_module && (bookings = staff_api.get_module_state(booking_module["id"].as_s).get["bookings"]?)
```

Comment:
You may need to expand the staff API driver to support this, but this can collapse into a single query via this endpoint:

https://github.com/PlaceOS/rest-api/blob/5d3acc0cff8d949d360a8f6de9c2e916c52f5cf1/src/placeos-rest-api/controllers/systems.cr#L314-L321